ods_ci/libs/DataSciencePipelinesAPI.py

```diff
@@ -0,0 +1,235 @@
+        print(f'create_operator({error}): {response}')
+
+    @keyword
+    def get_redhat_openshift_pipelines(self):
+        """
+        when creating at the first time, it can take like 1 minute to have the pods ready
+        """
+        deployment_count = 0
+        while deployment_count != 1:
+            deployments = []
+            response, _ = self.run_oc(f"oc get deployment -n openshift-operators openshift-pipelines-operator -o json")
+            try:
+                response = json.loads(response)
+                if response['metadata']['name'] == 'openshift-pipelines-operator' and 'readyReplicas' in response['status'] and response['status']['readyReplicas'] == 1:
+                    deployments.append(response)
+            except JSONDecodeError:
+                pass
+            deployment_count = len(deployments)
+            time.sleep(1)
+        return self.count_running_pods(f'oc get pods -n openshift-operators -l name=openshift-pipelines-operator -o json', 'openshift-pipelines-operator', 'Running', 1)
+
+    @keyword
+    def get_data_science_pipelines_operator_controller_manager(self):
+        """
+        when creating at the first time, it can take like 1 minute to have the pods ready
+        """
+        return self.count_running_pods(f'oc get pods -n redhat-ods-applications -o json', 'data-science-pipelines-operator-controller-manager', 'Running', 3)
+
+    @keyword
+    def login_using_user_and_password(self, user, pwd, project):
+        print("Fetch token")
+        basic_value = f'{user}:{pwd}'.encode('ASCII')
+        basic_value = base64.b64encode(basic_value).decode('ASCII')
+        response = requests.get(self.auth_url, headers={
+            'Authorization': f'Basic {basic_value}'
+        }, verify=False, allow_redirects=False)
+        url_with_token = response.headers['Location']
+        access_token_key = 'access_token='
+        idx_token = url_with_token.index(access_token_key) + len(access_token_key)
+        token = url_with_token[idx_token:]
+        idx_token = token.index('&')
+        self.sa_token = token[:idx_token]
+
+        print("Fetch the dsp route and verify it works")
+        while self.route == '':
+            self.route, _ = self.run_oc(f"oc get route -n {project} ds-pipeline-ui-sample --template={{{{.spec.host}}}}")
+            time.sleep(1)
+
+        status = -1
+        count = 0
+        while status != 200 and count < 30:
+            response, status = self.do_get(f'https://{self.route}/apis/v1beta1/runs', headers={
+                'Authorization': f'Bearer {self.sa_token}'
+            })
+            time.sleep(1)
+            count += 1
+            print(status)
+        return status
+
+    @keyword
+    def create_pipeline_project(self, project):
+        print(f"We are removing the project({project}) because we could run the test multiple times")
+        self.run_oc(f"oc delete project {project} --wait=true --force=true")
```

Comment:
wouldn't be safer to delete all the pipelines objects before deleting forcing the deletion of the project? It would ensure there are no dangling resources in the test cluster

Reply:
I believe we should investigate what resources are "running" that is preventing the project to be deleted. Also, removing all pipelines before removing the project seems like: "before removing the database we need to drop all tables"